smartsim/_core/_cli/build.py

```diff
@@ -412,50 +282,30 @@ def execute(
         else:
             logger.warning("Dragon installation failed")
 
-    try:
-        if not args.only_python_packages:
-            # REDIS/KeyDB
-            build_database(build_env, versions, keydb, verbose)
-
-            # REDISAI
-            build_redis_ai(
-                build_env,
-                versions,
-                device,
-                pt,
-                tf,
-                onnx,
-                args.torch_dir,
-                args.libtensorflow_dir,
-                verbose=verbose,
-                torch_with_mkl=args.torch_with_mkl,
-            )
-    except (SetupError, BuildError) as e:
-        logger.error(str(e))
-        return os.EX_SOFTWARE
+    # REDIS/KeyDB
+    build_database(build_env, versions, keydb, verbose)
+
+    if (CONFIG.lib_path / "redisai.so").exists():
+        logger.warning("RedisAI was previously built, run 'smart clean' to rebuild")
+    else:
+        build_redis_ai(current_platform, mlpackages, build_env, verbose)
 
     backends = installed_redisai_backends()
     backends_str = ", ".join(s.capitalize() for s in backends) if backends else "No"
-    logger.info(f"{backends_str} backend(s) built")
-
-    try:
-        if "torch" in backends:
-            check_py_torch_version(versions, device)
-        if "tensorflow" in backends:
-            check_py_tf_version(versions)
-        if "onnxruntime" in backends:
-            check_py_onnx_version(versions)
-    except (SetupError, BuildError) as e:
-        logger.error(str(e))
-        return os.EX_SOFTWARE
+    logger.info(f"{backends_str} backend(s) available")
+
+    if args.install_python_packages:
+        for package in mlpackages.values():
+            logger.info(f"Installing python packages for {package.name}")
+            package.pip_install(quiet=not verbose)
```

Comment:
Not entirely sure what you're suggesting here. In the example you gave, TF won't show up in `mlpackages` regardless of whether the user chose `--skip-tensorflow` or not. Are you saying that we should throw a warning since the user might expect that Tensorflow doesn't exist? If so, I suppose we could try to find the union of all ML packages across all the platforms in the config dir. Otherwise, I think we resigning ourselves to keeping a master list of supported packages which I'd prefer to avoid.

Reply:
> Are you saying that we should throw a warning since the user might expect that Tensorflow doesn't exist?

Exactly! My fear is that a user, on a linux machine, would do something to the nature of
```sh
$ smart build --device cpu
```
and they would then install and start using TF (backend and/or wheel).

They then, for whatever reason attempt to install SmartSim locally on their mac laptop using the same command, but then get confused when TF in silently not installed.

Its a very weird corner case, and I could be very easily talked into calling this "user error", especially if we have a compatibility matrix in the docs, but it seems like the kind of thing that _could_ happen and I wasn't sure how much we want to try and save the user from themselves at runtime.

Reply:
I think let's not try to catch it; it's a solvable problem (we could just compile tensorflow and provide it). Like I suggested above, I might be ok with computing (based on the JSON files in the config files) what all possible packages might be, but I think it's also going to get noisy.

So long story short, I think maybe we chalk it up to user error and point to the docs.